mlir/include/mlir/IR/Attributes.h

```diff
@@ -405,7 +405,7 @@ struct CastInfo<To, From,
   /// returns the dynamic ID. This means that T::classof would end up comparing
   /// the static TypeID of the children to the static TypeID of its parent,
   /// making it impossible to downcast from the parent to the child.
-  static inline bool isPossible(mlir::Attribute ty) {
+  static inline bool isPossible([[maybe_unused]] mlir::Attribute ty) {
```

Comment:
Seems like working around some gcc bugs? I don't think we should do that.

Reply:
See https://github.com/llvm/llvm-project/pull/68667
And also: https://github.com/llvm/llvm-project/pull/68528 ; maybe the guard in CMake isn't wide enough.

Reply:
Ah well that'll teach me to not search for issues/closed PRs before submitting a new one. In reality I just need to add the same CMake lines in my downstream builds (the place I was actually recently seeing the noise). Sorry!